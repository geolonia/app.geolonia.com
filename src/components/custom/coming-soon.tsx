import React from "react";
import styled from "styled-components";
import { __ } from "@wordpress/i18n";

type Props = {
  children: React.ReactNode;
  label?: string;
  style?: React.CSSProperties;
};

const Container = styled.div`
  position: relative;
`;

const Screen = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.6);
  color: white;
  font-size: 2em;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const ComingSoon = (props: Props) => {
  const label = props.label || __("Coming soon");
  return (
    <Container style={props.style}>
      {props.children}
      <Screen>
        <span>{label}</span>
      </Screen>
    </Container>
  );
};

export default ComingSoon;
