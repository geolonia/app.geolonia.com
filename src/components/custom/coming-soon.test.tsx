import React from 'react';
import { ComingSoon } from './coming-soon';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

test('it should render without crash', () => {
  const component = renderer.create(
    <ComingSoon>
      <div style={{ width: 100, height: 100 }}>
        <span>{'hello'}</span>
      </div>
    </ComingSoon>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('it should render the label and inner content', async () => {
  const result = render(
    <ComingSoon label={'coming soon!'}>
      <div style={{ width: 100, height: 100 }}>
        <span>{'hello'}</span>
      </div>
    </ComingSoon>,
  );
  const labels = await result.findAllByText('coming soon!');
  const contents = await result.findAllByText('hello');
  expect(labels).toHaveLength(1);
  expect(contents).toHaveLength(1);
});
