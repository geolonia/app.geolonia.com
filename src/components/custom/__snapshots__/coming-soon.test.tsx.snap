// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render without crash 1`] = `
<div
  className="sc-bdVaJa EXKLM"
>
  <div
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <span>
      hello
    </span>
  </div>
  <div
    className="sc-bwzfXH cpMATu"
  >
    <span>
      Coming soon
    </span>
  </div>
</div>
`;
